src/main.c

```diff
@@ -137,6 +148,24 @@ struct listener_ssl_config_t {
     } dynamic;
 };
 
+struct listener_ssl_config_t {
+    H2O_VECTOR(h2o_iovec_t) hostnames;
+    /**
+     * List of identities. First identity is the default and is used for handling ClientHello. Therefore it is guaranteed to contain
+     * a non-null SSL_CTX even when TLS below 1.3 is disabled. Rest of the identities are stored in the order of preference and do
+     * not have SSL_CTX.
+     */
```

Comment:
In commit aef7b4d, updated the design based on offline discussion with @mootoh.

Identities in the configuration file is listed by the order of preference.

The internal structure is kept as is, as stated in this doc-comment. The code becomes simpler by using 0 as the default internally and building the context for the default identity before others. This is because there are dependencies from other identities to the default (as the default one is used for receiving ClientHello, before choosing the best-matching identity).